app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User', foreign_key: :user_id
+  belongs_to :author, foreign_key: :author_id, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_create :update_posts_counter
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
```

Comment:
**[OPTIONAL]**

- [ ] Kindly make this method private because this method should not be accessed by the users directly 👍 😊